### PR TITLE
New point gesture

### PR DESCRIPTION
**Pull request by cpanican**
Added a new pointing up gesture. 

To generate custom datasets, create two folders of desired gesture. For instance, if you want to generate an "ok" gesture, create two folders namely OkImages and OkTest. Using `PalmTracker.py`, generate 1000 images on OkImages and 100 images on OkTest. We have to change the code manually in order to do that (for now).

After generating your own dataset, open `ModelTrainer.ipynb` and train the new gesture. Make sure to update the number of classes when adding/removing gestures. We currently have Swing, Palm, Fist, and Point. Therefore we have '4' below:
```
# second parameter in fully_connected is number of classes
convnet=fully_connected(convnet,4,activation='softmax')
```
`ModelTrainer.ipynb` exports the trained model in the `/ml/TrainedModel` directory.
PS: _50 epoch might be too much. Let's see if we can lower the epoch as we add more features._ 

Finally, update `ContinuousGesturePredictor.py` to match the number of classes we are classifying. Add a new if/else statement on `showStatistics()` function, and  update `convnet=fully_connected(convnet,4,activation='softmax')` similar to `ModelTrainer.ipynb`

### CODE REVIEW

File: ml/ContinuousGesturePredictor.py [hunk 1]
```diff
@@ -196,7 +196,8 @@ def showStatistics(predictedClass, confidence):
 convnet=fully_connected(convnet,1000,activation='relu')
 convnet=dropout(convnet,0.75)
 
-convnet=fully_connected(convnet,3,activation='softmax')
+# second parameter in fully_connected is number of classes
+convnet=fully_connected(convnet,4,activation='softmax')
 
 convnet=regression(convnet,optimizer='adam',learning_rate=0.001,loss='categorical_crossentropy',name='regression')
```
Comment (ayushyamitabh) [inline ml/ContinuousGesturePredictor.py:199]: Have we tried this with different optimizers? Is Adam the best for our case?
Same about learning rate

File: ml/ContinuousGesturePredictor.py [hunk 2]
```diff
@@ -151,6 +151,8 @@ def showStatistics(predictedClass, confidence):
         className = "Palm"
```
Comment (ayushyamitabh) [inline ml/ContinuousGesturePredictor.py]: Instead of adding if statements - let's change this to a dictionary with key - gesture name pairs

**Review by ayushyamitabh — CHANGES_REQUESTED** [review]
These can be separate PR's but additional comments -

1. Is there anyway to store video and then just have a script extract frames during run time rather than saving so many images separately - I can look into this. 

2. Make sure to a doc in the ML folder so we know how to test and train on our machines.

File: ml/ContinuousGesturePredictor.py (hunk 1 above)
Comment (cpanican) [reply ml/ContinuousGesturePredictor.py:199]: We haven't tried different optimizers. I believe Adam is one of the best optimizers out there according to some [sources](http://ruder.io/optimizing-gradient-descent/) however, [others](https://shaoanlu.wordpress.com/2017/05/29/sgd-all-which-one-is-the-best-optimizer-dogs-vs-cats-toy-experiment/) argue that it might not be as good. We should definitely play around with the parameters. One thing's for sure though, we can probably reduce our epoch from 50 to maybe around ~30 when training our model.

**Comment by cpanican** [comment]
I'll make the docs in a separate PR

**Comment by ayushyamitabh** [comment]
Great work! Let's keep the epoch and optimizer in mind and experiment with it going forward 😊
